### ISSUE TITLE
Temporary tsconfigs for a given project all use the same path.

### ISSUE THREAD

**Issue by statop** [nrwl/nx#21396]
### Current Behavior

All generated tsconfigs in the tmp folder for a given project have the path `tmp/<projectRoot>/tsconfig.generated.json`.

### Expected Behavior

All generated tsconfigs in the tmp folder for a given project have the path `tmp/<projectRoot>/<random_number>/tsconfig.generated.json`.

(or something like that)

### Steps to Reproduce

1. Create a project that has multiple tasks that generate tsconfigs. 
2. Note that if both tasks run concurrently they can conflict with each other.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 18.17.1
   OS     : darwin-arm64
   pnpm   : 8.14.3

   nx                 : 17.2.8
   @nx/js             : 17.2.8
   @nx/jest           : 17.2.8
   @nx/eslint         : 17.2.8
   @nx/workspace      : 17.2.8
   @nx/angular        : 17.2.8
   @nx/devkit         : 17.2.8
   @nx/eslint-plugin  : 17.2.8
   @nx/node           : 17.2.8
   @nx/plugin         : 17.2.8
   @nx/react          : 17.2.8
   @nx/rollup         : 17.2.8
   @nx/storybook      : 17.2.8
   @nx/vite           : 17.2.8
   @nx/web            : 17.2.8
   @nx/webpack        : 17.2.8
   typescript         : 5.2.2
   ---------------------------------------
   Community plugins:
   @openadp/tools-plugin : 24.5.0
   @storybook/angular    : 7.6.10
   ---------------------------------------
   Local workspace plugins:
   	 @openadp/tools-plugin
```

### Package Manager Version

pnpm  8.14.3

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Basically this is a problem with tsconfigs overwriting each other when run concurrently.
